src/reducers/index.ts

```diff
@@ -18,9 +42,64 @@ function reducer(state: Readonly<StateBase>, action: Action): StateBase {
   };
 }
 
+/**
+ * Exclude these actions from the history completely.
+ */
+const excludeActionsInHistory = [DATASET_INLINE_RECEIVE,
+  RESULT_RECEIVE,
+  RESULT_REQUEST,
+  SET_CONFIG,
+];
+
+/**
+ * A list of actions that can be initiated by a user.
+ *
+ * Each of these will start a new 'undo group'. Non-user actions will be put into the group
+ * of the preceding user action if one is available. If none is available it will be put
+ * into its own group.
+ */
+const userActions = [
+  // Dataset Actions
+  DATASET_URL_REQUEST,
+  DATASET_URL_RECEIVE,
+  DATASET_INLINE_RECEIVE,
```

Comment:
That is an oversight, it probably shouldn't be here. Again unless we can think of a reason why it should be here. At the moment i just can't think of one.